examples/rust/custom_store_view/src/main.rs

```diff
@@ -0,0 +1,201 @@
+//! This example demonstrates how to use [`StoreView`]s and [`StoreEvent`]s to implement both
+//! custom secondary indices and trigger systems.
+//!
+//! Usage:
+//! ```sh
+//! # Start the Rerun Viewer with our custom view in a terminal:
+//! $ cargo r -p custom_store_view
+//!
+//! # Log any kind of data from another terminal:
+//! $ cargo r -p objectron -- --connect
+//! ```
+
+use std::collections::BTreeMap;
+
+use rerun::{
+    external::{anyhow, re_arrow_store, re_build_info, re_log, re_log_types::TimeRange, tokio},
+    time::TimeInt,
+    ComponentName, EntityPath, StoreEvent, StoreId, StoreView, Timeline,
+};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<std::process::ExitCode> {
+    re_log::setup_native_logging();
+
+    let _handle = re_arrow_store::DataStore::register_view(Box::<Orchestrator>::default());
+    // Could use the returned handle to get a reference to the view if needed.
+
+    let build_info = re_build_info::build_info!();
+    rerun::run(build_info, rerun::CallSource::Cli, std::env::args())
+        .await
+        .map(std::process::ExitCode::from)
+}
+
+// ---
+
+/// A meta [`StoreView`] that distributes work to our other views.
+///
+/// The order is which registered views are executed is undefined: if you rely on a specific order
+/// of execution between your views, orchestrate it yourself!
+///
+/// Clears the terminal and resets the cursor for every new batch of [`StoreEvent`]s.
+#[derive(Default)]
+struct Orchestrator {
+    components_per_recording: ComponentsPerRecording,
+    time_ranges_per_entity: TimeRangesPerEntity,
+}
+
+impl StoreView for Orchestrator {
+    fn name(&self) -> String {
+        "rerun.store_view.ScreenClearer".into()
+    }
+
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+
+    fn on_events(&mut self, events: &[StoreEvent]) {
+        print!("\x1B[2J\x1B[1;1H"); // terminal clear + cursor reset
```

Comment:
I wonder if this will work on windows. It should. But I'm nervous. Will check later and fix for you if necessary :)